### ISSUE TITLE
Unexpected lines appear between items in a row

### ISSUE THREAD

**Issue by busslina** [flutter/flutter#137946]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1) main file:
```dart
import 'package:flutter/material.dart';
import 'package:general_test/pixel_issue/page.dart';

void main() {
  runApp(const App());
}

class App extends StatelessWidget {
  const App({super.key});

  @override
  Widget build(BuildContext context) => const MaterialApp(
        debugShowCheckedModeBanner: false,
        home: PageTable(),
      );
}
```

2) Page file:
```dart

import 'package:busslina_flutter_lightweight_lib/busslina_flutter_lightweight_lib.dart'
    as fllib;

import 'package:flutter/material.dart';

class PageTable extends StatelessWidget {
  const PageTable({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) => Row(
        children: [
          _buildExpandedChild(),
          _buildExpandedChild(),
          _buildExpandedChild(),
          _buildExpandedChild(),
        ],
      ).bgColor(Colors.black);

  Widget _buildExpandedChild() => Expanded(
        flex: 1,
        child: const fllib.Label(
          'Hey',
          textAlign: TextAlign.center,
          color: Colors.black,
        ).sized(height: double.infinity).bgColor(Colors.blue),
      );
}
```

### Expected results

No separation between items. Thought it could be a ´logical pixel´ issue, but it seems isn't, cause I tested with different screen/window sizes.

The lines always get the color of the parent background color.

Also mention that between the `2nd` and the `3rd` item, no line is shown.

Wrapping it in a `Scaffold` doesn't change the result.

### Actual results

![imagen](https://github.com/flutter/flutter/assets/73592852/00bb823b-052c-436c-940f-5d0744441edf)
![imagen](https://github.com/flutter/flutter/assets/73592852/304b53c3-39f6-4926-9b86-58e3b4521272)

**Comment by busslina** [comment]
> This is a duplicate of #14288
> 
> In short, Impeller fixes this for many cases on iOS, and will for Android once that is ready sometime next year.

1) Will be fixed also for Desktop (my case) and for web?
2) Can I use a temporary trick to avoid this waiting for the fix?

Thanks

**Comment by jonahwilliams** [comment]
1. Eventually, but it will take even longer.
2. If you read the thread, you'll find the package https://pub.dev/packages/pixel_snap linked which provides specific widgets to work around the issue

**Comment by busslina** [comment]
Thanks again
